src/App.tsx

```diff
@@ -1,223 +1,291 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import {
+  getTodos,
+  addTodos,
+  deleteTodos,
+  updateTodos,
+} from './api/todos';
+import { NewTodoField } from './components/NewTodoField/NewTodoField';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { FilterType } from './types/Filtertype';
+import { ErrorType } from './types/ErrorType';
+import { Error } from './components/Error/Error';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filterType, setFilterType] = useState(FilterType.all);
+  const [isAdding, setIsAdding] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [errorType, setErrorType] = useState(ErrorType.none);
+
+  const complitedTodos: Todo[] = todos.filter(todo => todo.completed);
+  const activeTodos: Todo[] = todos.filter(todo => !todo.completed);
+
+  const loadTodos = async () => {
+    try {
+      const result = await getTodos(user?.id || 1);
+
+      setTodos(result);
+    } catch {
+      setHasError(true);
+    }
+  };
+
+  useEffect(() => {
+    if (user) {
+      loadTodos();
+    }
+  }, []);
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
-  return (
-    <div className="todoapp">
-      <h1 className="todoapp__title">todos</h1>
+  useEffect(() => {
+    const timer = setTimeout(
+      () => {
+        setHasError(false);
+        clearTimeout(timer);
+      }, 3000,
+    );
+  }, [hasError]);
 
-      <div className="todoapp__content">
-        <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
+  const onFormSubmit = useCallback(
+    async (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      const queryTrim = query.trim();
 
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              ref={newTodoField}
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
-            />
-          </form>
-        </header>
+      if (queryTrim) {
+        setIsAdding(true);
+        try {
+          await addTodos({
+            userId: user?.id || +new Date(),
```

Comment:
If this ID is not there, then there is an error somewhere. There must be a user who performs these actions on the page

```suggestion
            userId: user?.id,
```